spec/controllers/hq/partners_controller_spec.rb

```diff
@@ -16,4 +16,29 @@ def foobar_action
       get :foobar_action
     end
   end
+
+  describe 'while signed-in' do
+    before :each do
+      sign_in instance_double(AdminUser)
+    end
+
+    describe 'GET #show' do
+      before :each do
+        FactoryGirl.create(:partner)
+      end
+
+      example 'invalid :id should redirect & set flash' do
+        get :show, id: (Partner.all.count + 1)
+        expect(response.status).to be 302
+        expect(response).to redirect_to hq_partners_path
+        expect(flash[:error]).to match /Cannot find Partner/
+      end
+
+      example 'valid :id should not redirect nor set flash' do
+        get :show, id: Partner.first.id
+        expect(response.status).to_not be 302
+        expect(flash[:error]).to_not match /Cannot find Partner/
+      end
+    end
```

Comment:
agreed in addition the presence of a passing happy path makes tthe check in this test unnecessary